test/acceptance/tests/mesh-gateway/mesh_gateway_test.go

```diff
@@ -0,0 +1,250 @@
+package meshgateway
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/hashicorp/consul-helm/test/acceptance/framework"
+	"github.com/hashicorp/consul-helm/test/acceptance/helpers"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Test that Connect and wan federation over mesh gateways work in a default installation
+// i.e. without ACLs because TLS is required for WAN federation over mesh gateways
+func TestMeshGatewayDefault(t *testing.T) {
+	env := suite.Environment()
+
+	primaryHelmValues := map[string]string{
+		"global.datacenter":                        "dc1",
+		"global.tls.enabled":                       "true",
+		"global.tls.httpsOnly":                     "false",
+		"global.federation.enabled":                "true",
+		"global.federation.createFederationSecret": "true",
+
+		"connectInject.enabled": "true",
+
+		"meshGateway.enabled":  "true",
+		"meshGateway.replicas": "1",
+	}
+
+	releaseName := helpers.RandomName()
+
+	// Install the primary consul cluster in the default kubernetes context
+	primaryConsulCluster := framework.NewHelmCluster(t, primaryHelmValues, env.DefaultContext(t), suite.Config(), releaseName)
+	primaryConsulCluster.Create(t)
+
+	secondaryContext := env.Context(t, framework.SecondaryContextName)
```

Comment:
We can extract the primary context as a var here as well so that it is easier to pass around instead of env.DefaultContext(t) 😃

Reply:
Done!